src/test/lib/ApiTest.cpp

```diff
@@ -1152,19 +1152,48 @@ DummyStreamCallback(
     return QUIC_STATUS_SUCCESS;
 }
 
+
 struct CloseFromCallbackContext {
     uint16_t CloseCount;
     volatile uint16_t CurrentCount;
+    // assuming server/client have 1 connection
+    MsQuicStream* ServerStreams[10] = {};
+    uint8_t NumServerStream {0};
+    MsQuicStream* ClientStreams[10] = {};
+    uint8_t NumClientStream {0};
+
+    uint8_t LRawBuffer[100];
+    QUIC_BUFFER LBuffer { sizeof(LRawBuffer), LRawBuffer };
+
+    static QUIC_STATUS StreamCallback(_In_ MsQuicStream* Stream, _In_opt_ void*, _Inout_ QUIC_STREAM_EVENT* Event) {
 
-    static QUIC_STATUS StreamCallback(_In_ MsQuicStream*, _In_opt_ void*, _Inout_ QUIC_STREAM_EVENT*) {
+        if (Event->Type == QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE) {
+            Stream->Close(); // or abort
+        }
         return QUIC_STATUS_SUCCESS;
     }
 
-    static QUIC_STATUS Callback(_In_ MsQuicConnection* Conn, _In_opt_ void* Context, _Inout_ QUIC_CONNECTION_EVENT* Event) {
+    static QUIC_STATUS Callback(_In_ MsQuicConnection* Conn, _In_opt_ void* Context, _Inout_ QUIC_CONNECTION_EVENT* Event, bool IsServer) {
         auto Ctx = (CloseFromCallbackContext*)Context;
 
         if (Event->Type == QUIC_CONNECTION_EVENT_PEER_STREAM_STARTED) {
-            new(std::nothrow) MsQuicStream(Event->PEER_STREAM_STARTED.Stream, CleanUpAutoDelete, StreamCallback, nullptr);
+            // This stream send shutdown by receiving FIN
+            new(std::nothrow) MsQuicStream(Event->PEER_STREAM_STARTED.Stream, CleanUpAutoDelete, StreamCallback, Context);
+        }
+        if (IsServer) {
+            if (Event->Type == QUIC_CONNECTION_EVENT_CONNECTED) {
+                QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
+
+                MsQuicStream* Stream = new(std::nothrow) MsQuicStream(*Conn, QUIC_STREAM_OPEN_FLAG_UNIDIRECTIONAL, CleanUpAutoDelete, StreamCallback, Context);
+                Ctx->ServerStreams[Ctx->NumServerStream++] = Stream; // to delete if connection is deleted earlier
+                Stream->Start(QUIC_STREAM_START_FLAG_NONE);
```

Comment:
The `Start` call is unnecessary since you are passing the `QUIC_SEND_FLAG_START` in the next call.